massa-network-worker/src/messages.rs

```diff
@@ -57,12 +57,20 @@ pub enum Message {
     Endorsements(Vec<SignedEndorsement>),
 }
 
-/// Deserialize, and return, a message.
-/// In the case of a block,
-/// also return the serialized object.
+/// The serialized form of an object, as received from the network.
+#[derive(Debug)]
+pub enum SerializedForm {
+    /// A serialized block.
+    Block(Vec<u8>),
+    /// A serialized list of operations.
+    Operations(Vec<Vec<u8>>),
+}
+
+/// Deserialize, and return, a message,
+/// while retaining the serialized format of the object(s) contained in it.
 pub fn deserialize_message_with_optional_serialized_object(
     buffer: &[u8],
-) -> Result<(Message, Option<Vec<u8>>), ModelsError> {
+) -> Result<(Message, Option<SerializedForm>), ModelsError> {
```

Comment:
I think it's kind of the same as wrapping `SerializedForm` inside an option as now, actually I think it's clearer because having a `None` variant could mean "does not have a serialized form", where in fact it does, but we just don't use it.

Reply:
Yes, indeed it's more idiomatic. I just think at loud about the format _Enum in Enum_